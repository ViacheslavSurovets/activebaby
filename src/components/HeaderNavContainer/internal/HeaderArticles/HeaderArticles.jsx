import React, { useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { DropdownGroupArticle, DropdownLink, ListItem } from '@components/Header/styles';
import Proptypes from 'prop-types';

const HeaderArticles = ( { headerArticlesData } ) => {
  const { t, i18n } = useTranslation ();

  const headerArticles = useMemo ( () => (
    headerArticlesData.map ( ( item, idx ) => {
      return {
        key: idx,
        text: t ( item.text ),
        path: item.path
      };
    } )

  ), [ i18n.language ] );

  return (
    <>
      { headerArticles.map ( item =>
        <DropdownGroupArticle key={ item.text }>
          <ListItem key={ item.text }><DropdownLink
            to={ item.path }>{ item.text }</DropdownLink></ListItem>
        </DropdownGroupArticle>
      ) }
    </>
  );
};

export default HeaderArticles;

HeaderArticles.propTypes = {
  headerArticlesData: Proptypes.array.isRequired
};

