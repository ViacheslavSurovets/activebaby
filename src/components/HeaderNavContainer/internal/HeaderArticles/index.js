export { default as HeaderArticles } from './HeaderArticles';
