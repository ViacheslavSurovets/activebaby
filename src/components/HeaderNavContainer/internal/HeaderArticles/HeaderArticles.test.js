import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import HeaderArticles from './HeaderArticles';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('HeaderArticles', ()=> {
  it('HeaderArticles Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <HeaderArticles  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
