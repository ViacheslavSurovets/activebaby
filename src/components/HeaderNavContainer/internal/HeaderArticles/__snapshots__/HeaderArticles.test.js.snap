// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderArticles HeaderArticles Snapshot 1`] = `ShallowWrapper {}`;
